Sources/Routing/RouteBuilder.swift

```diff
@@ -6,24 +6,25 @@ public typealias WebSocketRouteHandler = (Request, WebSocket) throws -> Void
 
 /// Used to define behavior of objects capable of building routes
 public protocol RouteBuilder: class {
-    func register(host: String?, method: Method, path: [String], responder: Responder)
+    func register(host: String?, method: Method, path: [String], metadata: [String: String]?, responder: Responder)
```

Comment:
Changing this signature is definitely a breaking change.

Reply:
vapor/vapor#1463 would be required, i wasn't sure how to go about it so I included it in the description above.

Reply:
Other people relying on this API will still notice the breaking change

Reply:
So changing the protocol in any way wouldn't be an option I take it. Any thoughts? I was going to try and implement a wrapper around `RouteBuilder` but that still wouldn't allow me to filter the metadata all the way up the `register` chain and to the `Router`.

Reply:
A second routing protocol is the only thing I could come up with. The current Vapor 2 routers could conform to both routing protocols. That way current users' don't find a breaking change, but the features are still there.

Maybe someone else has a more creative solution